packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -601,9 +589,77 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
         result.append(startNode)
         unresolved.remove(startNode)
     }
-    
+
     // TODO: Remove once all configure is moved to the bridge
     func getPlugin() throws -> AWSDataStorePlugin {
         return try Amplify.DataStore.getPlugin(for: "awsDataStorePlugin") as! AWSDataStorePlugin
     }
+    
+    func createErrorHandler(args: [String: Any])  -> DataStoreErrorHandler {
+        var errorHandler: DataStoreErrorHandler
+        if((args["hasErrorHandler"] as? Bool) == true) {
+            errorHandler = { error in
+                let map : [String:Any] = [
+                    "errorCode" : "DataStoreException",
+                    "errorMesage" : ErrorMessages.defaultFallbackErrorMessage,
+                    "details" : FlutterDataStoreErrorHandler.createSerializedError(error: error)
+                ]
+                self.channel!.invokeMethod("errorHandler", arguments: map)
+            }
+        } else {
+            errorHandler = { error in
+                Amplify.Logging.error(error: error)
+            }
+        }
+        return errorHandler
+    }
+    
+    func createConflictHandler(args: [String: Any])  -> DataStoreConflictHandler {
+        var conflictHandler:DataStoreConflictHandler
+        if((args["hasConflictHandler"] as? Bool) == true) {
+            conflictHandler = { conflictData, onDecision in
+                do {
+                    let modelName = conflictData.remote["__typename"] as! String
+                    let local = conflictData.local as! FlutterSerializedModel
+                    let remote = conflictData.remote as! FlutterSerializedModel
+
+                    let map : [String:Any] = [
+                        "modelName" : modelName,
+                        "local" : try local.toMap(flutterModelRegistration: self.modelSchemaRegistry, modelName: modelName),
+                        "remote" : try remote.toMap(flutterModelRegistration: self.modelSchemaRegistry, modelName: modelName)
+                    ]
+
+                    self.channel!.invokeMethod("conflictHandler", arguments: map){ result in
+                        do {
+                            let resultMap : [String: Any] = result as! [String: Any]
+                            switch(resultMap["resolutionStrategy"] as! String){
+                                case "APPLY_REMOTE": onDecision(.applyRemote)
```

Comment:
Ditto can we use exiting enum if any?

Reply:
Hmm I don't think this will be so straightforward to do.  

I don't know how to directly create an enum from a string without declaring the enum with : string.  (see [stackoverflow](https://stackoverflow.com/questions/30009788/in-swift-is-it-possible-to-convert-a-string-to-an-enum/30009873)) 

In the example implementation I used above, the code fails: 
```

import Foundation

public protocol Model: Codable {
    var modelName: String { get }
}

public enum DataStoreConflictHandlerResult {

    /// Discard the local changes in favor of the remote ones. Semantically the same as `DISCARD` on Amplify-JS
    case applyRemote

    /// Keep the local changes. (semantic shortcut to `retry(local)`).
    case retryLocal

    /// Return a new `Model` instance that should used instead of the local and remote changes.
    case retry(Model)
}

print( DataStoreConflictHandlerResult(rawValue: "retry") )
```